src/main/php/cfg/verb/verb.php

```diff
@@ -756,140 +758,144 @@ private function save_field_do(sql_db $db_con, $log): string
     {
-        $result = '';
+        $usr_msg = null;
         if ($db_rec->rev_plural <> $this->rev_plural) {
             $log = $this->log_upd();
             $log->old_value = $db_rec->rev_plural;
             $log->new_value = $this->rev_plural;
             $log->std_value = $db_rec->rev_plural;
             $log->row_id = $this->id();
             $log->set_field(self::FLD_PLURAL_REVERSE);
-            $result .= $this->save_field_do($db_con, $log);
+            $usr_msg = $this->save_field_do($db_con, $log);
         }
-        return $result;
+        return $usr_msg;
     }
 
     // set the update parameters for the verb description
-    private function save_field_description(sql_db $db_con, $db_rec): string
+    private function save_field_description(sql_db $db_con, $db_rec): user_message
     {
-        $result = '';
+        $usr_msg = null;
         if ($db_rec->description <> $this->description) {
             $log = $this->log_upd();
             $log->old_value = $db_rec->description;
             $log->new_value = $this->description;
             $log->std_value = $db_rec->description;
             $log->row_id = $this->id();
             $log->set_field(sandbox_named::FLD_DESCRIPTION);
-            $result .= $this->save_field_do($db_con, $log);
+            $usr_msg = $this->save_field_do($db_con, $log);
         }
-        return $result;
+        return $usr_msg;
     }
 
     // set the update parameters for the verb description
-    private function save_field_formula_name(sql_db $db_con, $db_rec): string
+    private function save_field_formula_name(sql_db $db_con, $db_rec): user_message
     {
-        $result = '';
+        $usr_msg = null;
         if ($db_rec->frm_name <> $this->frm_name) {
             $log = $this->log_upd();
             $log->old_value = $db_rec->frm_name;
             $log->new_value = $this->frm_name;
             $log->std_value = $db_rec->frm_name;
             $log->row_id = $this->id();
             $log->set_field(self::FLD_FORMULA);
-            $result .= $this->save_field_do($db_con, $log);
+            $usr_msg = $this->save_field_do($db_con, $log);
         }
-        return $result;
+        return $usr_msg;
     }
 
     // save all updated verb fields excluding the name, because already done when adding a verb
-    private function save_fields(sql_db $db_con, $db_rec): string
+    private function save_fields(sql_db $db_con, $db_rec): array
     {
-        $result = $this->save_field_code_id($db_con, $db_rec);
-        $result .= $this->save_field_plural($db_con, $db_rec);
-        $result .= $this->save_field_reverse($db_con, $db_rec);
-        $result .= $this->save_field_rev_plural($db_con, $db_rec);
-        $result .= $this->save_field_description($db_con, $db_rec);
-        $result .= $this->save_field_formula_name($db_con, $db_rec);
+        $usr_msg[] = $this->save_field_code_id($db_con, $db_rec);
```

Comment:
I guess, you need to use the $usr_msg->add() function here, because $usr_msg is supposed to be an object not a simple array.